### PR TITLE
Made it csv

### PR DESCRIPTION
**Pull request by kyleburgess2025**

### CODE REVIEW

File: lib/modules/databasemodule/database.py [hunk 1]
```diff
@@ -355,6 +355,15 @@ def remove_intern(self, id):
         )
         self.con.commit()
         logging.info("Intern removed from intern table")
+    
+    def return_all_interns(self):
+        """Returns all intern information in the table."""
+
+        self.cur.execute(
+            f"SELECT * FROM intern"
```
Comment (lunstb) [inline lib/modules/databasemodule/database.py:363]: why is this an f string?

File: lib/modules/internmodule/internmodule.py [hunk 2]
```diff
@@ -37,10 +37,9 @@ def interpret_message(self, msg, email, db, admin) -> Dict:
                 else:
                     return catch_incorrect_arguments(Commands.INTERN_REMOVE)
             if commands[0] == 'list':
-                if len(commands) <= 2:
+                if len(commands) == 1:
```
Comment (lunstb) [inline lib/modules/internmodule/internmodule.py:40]: this code is confusing, is there a way you could easily retructure it to be simpler?

File: lib/modules/internmodule/internresponse.py [hunk 3]
```diff
@@ -20,40 +29,29 @@ def not_removed_intern(error):
...
-        response += f"{user_name} works at {company} as a {position} and is {answer}able to give references. " + resume_answer + f" Email them at {user_email}.\n"
-
-    return response
\ No newline at end of file
+def list_interns(db: database, channel_id):
+    """Returns a csv with intern data"""
+
+    all_intern_info = db.return_all_interns()
+    with open("out.csv", "w", newline='') as csv_file:
+        csv_writer = csv.writer(csv_file)
```
Comment (lunstb) [inline lib/modules/internmodule/internresponse.py:37]: There are some really cool csv libraries in python. Notably if you have a  list of dictionaries or something like that (you may be able to get them from return all interns) you can very easily turn that into a csv. Logic might be simpler

File: lib/modules/internmodule/internresponse.py [hunk 4]
```diff
@@ -20,40 +29,29 @@ def not_removed_intern(error):
...
+                resume_link = self_resumes[0][1]
+            else:
+                resume_link = "N/A"
+            csv_writer.writerow([i[0], i[1], i[2], i[3], i[4], resume_link])
+    my_file = {'file' : ('out.csv', open('out.csv', 'rb'), 'csv')}
+    payload = {
+        'filename': 'intern_data.csv',
+        'channels': channel_id
+        }
+    r = requests.post("https://slack.com/api/files.upload", params=payload, files=my_file, headers={'Authorization': 'Bearer ' + os.getenv("SLACK_BOT_TOKEN")}).content
```
Comment (lunstb) [inline lib/modules/internmodule/internresponse.py:52]: urls are happy in constant places 🗡️ 😄

File: out.csv [hunk 5]
```diff
@@ -0,0 +1,2 @@
+Name,Email,Company,Position,Able to Give Referrals?,Resume
```
Comment (lunstb) [inline out.csv:1]: Let's try to put this in gitignore? also do we want this file saved or just sent back? thoughts?

**Review by lunstb — CHANGES_REQUESTED** [review]

File: lib/modules/internmodule/internresponse.py (hunk 3 above)
Comment (kyleburgess2025) [reply lib/modules/internmodule/internresponse.py:37]: I will look into this further later - I think this is good for now

**Pull request merged**